syntax = "proto3";

package myapp;

message Book {
    int32 id = 1;
    string title = 2;
    string author = 3;
}

message BookRequest {
    int32 id = 1;
}

message Empty {}

message ListBooksRequest {}

message ListBooksResponse {
    repeated Book books = 1;
}

service BookService {
    rpc GetBookById(BookRequest) returns (Book);
    rpc GetAllBooks(Empty) returns (ListBooksResponse);
}
